src/common/signalk/SKNMEA2000Parser.cpp

```diff
@@ -43,19 +46,170 @@ const SKUpdate& SKNMEA2000Parser::parse(const SKSourceInput& input, const tN2kMs
     delete(_sku);
   }
 
-  // Boat speed
-  if (msg.PGN == 128259) {
-    return parse128259(input, msg, timestamp);
+  DEBUG("N2k Message PGN: %i", msg.PGN);
+  switch (msg.PGN) {
+    case 126992L: // System Time / Date
+      //if ( cfEnableSystemDateTimeFromPGN_126992 ) {
+        return parse126992(input, msg, timestamp);
+      //}
+      break;
+    case 127245L: // Rudder
+      //if ( cfEnableN2kToNMEA_127245 ) {
+        DEBUG("Rudder message from NMEA2000");
+        return parse127245(input, msg, timestamp);
+      //}
+      break;
+    case 127250L: // Vessel Heading
+      //if ( cfEnableN2kToNMEA_127250 ) {
+        return parse127250(input, msg, timestamp);
+      //}
+      break;
+    case 128259L: // Boat speed
+      //if ( cfEnableN2kToNmea_128259 ) {
+        return parse128259(input, msg, timestamp);
+      //}
+      break;
+    case 128267L: // Water depth
+      //if ( cfEnableN2kToNMEA_128267 ) {
+        return parse128267(input, msg, timestamp);
+      //}
+      break;
+    case 129026L: // COG SOG rapid
+      //if ( cfEnableN2kToNMEA_129026 ) {
+        return parse129026(input, msg, timestamp);
+      //}
+      break;
+    case 130306L: // Wind Speed
+      //if ( cfEnableN2kToNMEA_130306 ) {
+        return parse130306(input, msg, timestamp);
+      //}
+      break;
+
+    //case 126993: // Heartbeat
+    //case 127251: // Rate of Turn
+    //case 127257: // Attitude
+    //case 127488: // Engine parameters rapid
+    //case 127493: // Transmission parameters: dynamic
+    //case 127501: // Binary status report
+    //case 127505: // Fluid level
+    //case 127508: // Battery Status
+    //case 127513: // Battery Configuration Status
+    //case 129025: // Lat/lon rapid
+    //case 129283: // Cross Track Error
+    //case 130310: // Outside Environmental parameters
+    //case 130311: // Environmental parameters
+    //case 130312: // Temperature
+    //case 130314: // Pressure
+    //case 130316: // Temperature extended range
+    default:
+      DEBUG("No known conversion for PGN %i", msg.PGN);
+      return _invalidSku;
+  } // end switch msg.PGN
+}
+
+// System Time / Date
+// TODO: Set system date and time for SD-Card logging file date and
+// Offset to millis() for recalculation of true date/time
+// TODO: after 42 days millis() overflow, new time sync needed
```

Comment:
KBox has a real-time clock: https://www.pjrc.com/teensy/td_libs_Time.html#teensy3. This works on your teensy 3.6 too. Just need to add a crystal on the board.

Reply:
I will investigate this.
I think a battery for keeping time is not so good (as it is not in KBox too) and the power for navigation is turned off in the marina.
GPS from NMEA and/or N2k is fine.

Reply:
The RTC is independent of a battery. The battery just means that the time is kept when you power off.

With or without a battery, the RTC is a better way to measure time than the `millis()` function. It will be much more precise (because it uses a 32khz quartz) and it is simpler for us because the hardware does it automatically.